### PR TITLE
[spdk-isal] only linux

### PR DESCRIPTION
**Pull request by autoantwort**

### CODE REVIEW

File: scripts/ci.baseline.txt [hunk 1]
```diff
@@ -1408,14 +1408,10 @@ spdk-ipsec:x64-windows=fail
 spdk-ipsec:x64-windows-static=fail
 spdk-ipsec:x64-windows-static-md=fail
 spdk-ipsec:x86-windows=fail
-spdk-isal:arm64-windows=fail
-spdk-isal:arm-uwp=fail
-spdk-isal:x64-osx=fail
 spdk-isal:x64-uwp=fail
 spdk-isal:x64-windows=fail
 spdk-isal:x64-windows-static=fail
 spdk-isal:x64-windows-static-md=fail
```
Comment (JonLiu1993) [inline scripts/ci.baseline.txt:1414]: @autoantwort ,Thanks for your pr,  could you please clarify why these four items have not been deleted?

Comment (JonLiu1993) [inline scripts/ci.baseline.txt:1414]: @autoantwort ,Thanks for your pr, could you please clarify why these four items have not been deleted?

Comment (autoantwort) [reply scripts/ci.baseline.txt:1414]: Are now deleted. In general this port only builds in linux and now windows ...

**Comment by JonLiu1993** [comment]
@autoantwort, Looks the ci build on x64-linux,this is the error log:

````
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:155 (message):
    Command failed: /mnt/vcpkg-ci/downloads/tools/cmake-3.21.1-linux/cmake-3.21.1-linux-x86_64/bin/cmake --build . --config Debug --target install -- -v -j33
    Working Directory: /mnt/vcpkg-ci/buildtrees/spdk-isal/x64-linux-dbg
    See logs for more information:
      /mnt/vcpkg-ci/buildtrees/spdk-isal/install-x64-linux-dbg-out.log

Call Stack (most recent call first):
````
/mnt/vcpkg-ci/buildtrees/spdk-isal/install-x64-linux-dbg-out.log
````
[cppgraphqlgen] Update from v4.0.0 to v4.1.0[1/2] cd /mnt/vcpkg-ci/buildtrees/spdk-isal/src/spdk-d34ebb51cd.clean && -f
FAILED: CMakeFiles/isal /mnt/vcpkg-ci/buildtrees/spdk-isal/x64-linux-dbg/CMakeFiles/isal 
cd /mnt/vcpkg-ci/buildtrees/spdk-isal/src/spdk-d34ebb51cd.clean && -f
/bin/sh: 1: -f: not found
ninja: build stopped: subcommand failed.
````

**Review by JonLiu1993 — APPROVED** [review]

File: ports/spdk-isal/portfile.cmake [hunk 2]
```diff
@@ -1,48 +1,26 @@
...
+vcpkg_find_acquire_program(YASM)
 
-VCPKG_CONFIGURE_CMAKE(
-        SOURCE_PATH ${CMAKE_CURRENT_LIST_DIR}
-        PREFER_NINJA
-        OPTIONS
-        -DSOURCE_PATH=${SOURCE_PATH}
-        -DEXEC_ENV:STRING=${EXEC_ENV}
-        -DLIBRARY_LINKAGE:STRING=${VCPKG_LIBRARY_LINKAGE}
+vcpkg_configure_make(
```
Comment (JackBoosY) [inline ports/spdk-isal/portfile.cmake:16]: I think it also support OSX since it use makefile, can you please confirm that?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/spdk-isal/portfile.cmake (hunk 2 above)
Comment (autoantwort) [reply ports/spdk-isal/portfile.cmake:16]: It fails to build with 
```
erasure_code/gf_vect_dot_prod_sse.s:198: error: unrecognised symbol type `function'
erasure_code/gf_vect_mul_sse.s:116: error: unrecognised symbol type `function'
make[1]: *** [erasure_code/gf_vect_dot_prod_sse.lo] Error 1
make[1]: *** Waiting for unfinished jobs....
make[1]: *** [erasure_code/gf_vect_mul_sse.lo] Error 1
erasure_code/gf_vect_mul_avx.s:115: error: unrecognised symbol type `function'
make[1]: *** [erasure_code/gf_vect_mul_avx.lo] Error 1
erasure_code/gf_vect_dot_prod_avx.s:198: error: unrecognised symbol type `function'
make[1]: *** [erasure_code/gf_vect_dot_prod_avx.lo] Error 1
erasure_code/gf_vect_dot_prod_avx2.s:206: error: unrecognised symbol type `function'
make[1]: *** [erasure_code/gf_vect_dot_prod_avx2.lo] Error 1
erasure_code/gf_2vect_dot_prod_sse.s:243: error: unrecognised symbol type `function'
make[1]: *** [erasure_code/gf_2vect_dot_prod_sse.lo] Error 1
erasure_code/gf_3vect_dot_prod_sse.s:265: error: unrecognised symbol type `function'
make[1]: *** [erasure_code/gf_3vect_dot_prod_sse.lo] Error 1
erasure_code/gf_4vect_dot_prod_sse.s:298: error: unrecognised symbol type `function'
make[1]: *** [erasure_code/gf_4vect_dot_prod_sse.lo] Error 1
make: *** [all] Error 2
```

File: ports/spdk-isal/portfile.cmake [hunk 3]
```diff
@@ -1,48 +1,26 @@
...
+    SOURCE_PATH "${SOURCE_PATH}"
```
Comment (JackBoosY) [inline ports/spdk-isal/portfile.cmake:17]: Can you please also check the options in https://github.com/spdk/spdk/blob/master/configure#L28-L104?

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/spdk-isal/portfile.cmake (hunk 3 above)
Comment (autoantwort) [reply ports/spdk-isal/portfile.cmake:17]: Why? It is another repo?

Comment (JackBoosY) [reply ports/spdk-isal/portfile.cmake:17]: Sorry, I picked the wrong repo.

**Review by JackBoosY — APPROVED** [review]

**Comment by BillyONeal** [comment]
Wow I'm getting very https://github.com/microsoft/vcpkg/pull/22269/ vibes! Thanks!

**Pull request merged**